drivers/xen/xengnttab/arch/x86_64/gnttab.c

```diff
@@ -58,3 +66,62 @@ grant_entry_v1_t *gnttab_arch_init(int grant_frames_num)
 out:
 	return gnte;
 }
+#else
+
+/* Find memory location where `pages` pages can be added to
+ * an unallocated space */
+void *gnttab_reserve_memory(unsigned int pages)
+{
+	struct ukplat_memregion_desc *mrd;
+	__sz previous_region_end = 0;
+	__sz size = (unsigned long) pages * PAGE_SIZE;
+
+	ukplat_memregion_foreach(&mrd, 0, 0, 0) {
+		if (previous_region_end == 0) {
+			previous_region_end = mrd->pbase + mrd->pg_count * PAGE_SIZE;
+			continue;
+		}
+
+		if (previous_region_end + size <= mrd->pbase)
+			return (void *) previous_region_end;
+
+		previous_region_end = mrd->pbase + mrd->len;
+	}
+	/* The address must be a valid 32bit address as we are unsing gnttab v1
+	 * entries */
```

Comment:
there is no restriction regarding grant table placement (including with v1), though this restriction applies to grant entries which requires to be under 4GB in this version

Ideally, unikraft should use grant table v2 (with eventual optional support for v1), so that you wont have to bother about grant-able pages. I realistically don't expect grant table v1 (thus v2 disabled) to be forced onto a unikraft guest.